File: Year II/First Semester/MAP/Laborator/Laborator9/ConsoleApp1/ConsoleApp1/Model/MessageTask.cs
namespace ConsoleApp1.Model;

public class MessageTask: Task
{
    private string Message { get; set; }
    private string From { get; set; }
    private string To { get; set; }
    private DateTime Date { get; set; }
    
    public MessageTask(string taskId, string taskDescription, string message, string from, string to, DateTime date): base(taskId, taskDescription)
    {
        Message = message;
        From = from;
        To = to;
        Date = date;
    }

    public override void Execute()
    {
        Console.WriteLine($"\nFrom: {From} To: {To}\nMessage: {Message}\nDate: {Date}");
    }
}

File: Year II/First Semester/MAP/Seminar/seminar1/seminar1/src/model/MessageTask.java
package model;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class MessageTask extends Task {
    private String message;
    private String from;
    private String to;
    private LocalDateTime date;

    private static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");

    public MessageTask(String message, String from, String to, LocalDateTime date, String taskId, String description){
        super(taskId, description);
        this.message = message;
        this.from = from;
        this.to = to;
        this.date = date;
    }

    @Override
    public void execute() {
        System.out.println("mesaj="+message+"| date="+date.format(formatter));

    }
    //id=1 | description=Feedback lab1| message=Ai obtinut 9.60 | from=Maria | to=Ana | date=2018-09-27 09:29

    public String toString(){
        return "id="+this.getTaskId() +"|description=" + this.getDescription() + "|message=" + message + "|from="
                + from + "|to =" + to + "|date =" + date.format(formatter);
    }
}